src/Panel.tsx

```diff
@@ -85,20 +88,38 @@ const CollapsePanel = React.forwardRef<HTMLDivElement, CollapsePanelProps>((prop
     ...(['header', 'icon'].includes(collapsible) ? {} : collapsibleProps),
   };
 
+  const HeaderWrapper = ({ children: headerWrapperChildren }: PropsWithChildren) => {
+    if (!headingLevel) {
+      return <>{headerWrapperChildren}</>;
+    } else {
+      return (
+        <div className={`${prefixCls}-header-wrapper`} role="heading" aria-level={headingLevel}>
+          {headerWrapperChildren}
+        </div>
+      );
+    }
+  };
+
   // ======================== Render ========================
   return (
-    <div {...resetProps} ref={ref} className={collapsePanelClassNames}>
-      <div {...headerProps}>
-        {showArrow && iconNode}
-        <span
-          className={classNames(`${prefixCls}-title`, customizeClassNames?.title)}
-          style={styles?.title}
-          {...(collapsible === 'header' ? collapsibleProps : {})}
+    <div {...resetProps} ref={ref} className={collapsePanelClassNames} id={id}>
+      <HeaderWrapper>
+        <div
+          {...headerProps}
+          id={id ? `${id}__header` : undefined}
```

Comment:
Since `id` is provided by parent. So this will be the same `${id}__header` for all the Panel which is not correct. You can just `useId` to get own id instead.